code/notificationbuilder/src/main/java/com/adobe/marketing/mobile/notificationbuilder/internal/builders/ProductCatalogNotificationBuilder.kt

```diff
@@ -0,0 +1,335 @@
+     * @param expandedLayout the [RemoteViews] object containing the expanded layout of the push template notification
+     * @param catalogItems the list of [ProductCatalogPushTemplate.CatalogItem] objects containing the product catalog items
+     * @param pushTemplate the [ProductCatalogPushTemplate] object containing the product catalog push template data
+     */
+    private fun populateThumbnails(
+        context: Context,
+        broadcastReceiverClass: Class<out BroadcastReceiver>?,
+        channelIdToUse: String,
+        expandedLayout: RemoteViews,
+        catalogItems: List<ProductCatalogPushTemplate.CatalogItem>,
+        pushTemplate: ProductCatalogPushTemplate
+    ) {
+        Log.trace(
+            LOG_TAG,
+            SELF_TAG,
+            "Populating product catalog thumbnails."
+        )
+
+        val thumbIds = listOf(
+            R.id.product_thumbnail_1,
+            R.id.product_thumbnail_2,
+            R.id.product_thumbnail_3
+        )
+        for (index in catalogItems.indices) {
+            val thumbImage = PushTemplateImageUtils.getCachedImage(catalogItems[index].img)
+
+            if (thumbImage == null) {
+                Log.trace(
+                    LOG_TAG,
+                    SELF_TAG,
+                    "No image found for catalog item thumbnail."
+                )
+            } else {
+                expandedLayout.setImageViewBitmap(thumbIds[index], thumbImage)
+            }
+
+            // create a pending intent for the thumbnail
+            val pendingIntentThumbnailInteraction = createThumbnailInteractionPendingIntent(
+                context,
+                broadcastReceiverClass,
+                channelIdToUse,
+                pushTemplate,
+                index
+            )
+            expandedLayout.setOnClickPendingIntent(
+                thumbIds[index],
+                pendingIntentThumbnailInteraction
+            )
+        }
+    }
+
+    /**
+     * Sets the CTA button for the product catalog notification.
+     *
+     * @param context the application [Context]
+     * @param trackerActivityClass the [Class] of the activity to set in the created pending intent for tracking purposes
+     * @param expandedLayout the [RemoteViews] object containing the expanded layout of the push template notification
+     * @param pushTemplate the [ProductCatalogPushTemplate] object containing the product catalog push template data
+     */
+    private fun setupCtaButton(
+        context: Context,
+        trackerActivityClass: Class<out Activity>?,
+        expandedLayout: RemoteViews,
+        pushTemplate: ProductCatalogPushTemplate
+    ) {
+        // apply the color to the cta button
+        expandedLayout.setCtaButtonColor(R.id.cta_button, pushTemplate.ctaButtonColor)
+
+        // apply text to the cta button
+        expandedLayout.setTextViewText(R.id.cta_button, pushTemplate.ctaButtonText)
```

Comment:
Do we need to also accept and set the cta button text color?

Reply:
cta button text color is not configurable

Reply:
i imagine it would be good to have the option though

Reply:
We can add a key for it if you think it would be useful since we allow configuring the cta button background color

Reply:
i'll add a key and will update the wiki

Reply:
added `adb_cta_txt_clr` for cta text color customization